src/serialization/Portable.ts

```diff
@@ -41,127 +48,607 @@ export enum FieldType {
+     * @deprecated since version 4.2 for the sake of better naming. Please use {@link readString} instead.
+     * This method will be removed in next major version.
      */
-    readUTF(fieldName: string): string;
+    readUTF(fieldName: string): string | null;
 
-    readString(fieldName: string): string;
+    /**
+     * Reads a string from UTF-8 encoded bytes.
+     *
+     * @param fieldName name of the field
+     * @return the string value read
+     */
+    readString(fieldName: string): string | null;
 
+    /**
+     * Reads a primitive boolean.
+     *
+     * @param fieldName name of the field
+     * @return the boolean value read
+     */
     readBoolean(fieldName: string): boolean;
 
+    /**
+     * Reads a 8-bit unsigned integer.
+     *
+     * @param fieldName name of the field
+     * @return the byte value read
+     */
     readByte(fieldName: string): number;
 
+    /**
+     * Reads a single character string using `String.fromCharCode` from two bytes of UTF-16 code units.
+     *
+     * @param fieldName name of the field
+     * @return the char value read
+     */
     readChar(fieldName: string): string;
 
+    /**
+     * Reads a double.
+     *
+     * @param fieldName name of the field
+     * @return the double value read
+     */
     readDouble(fieldName: string): number;
 
+    /**
+     * Reads a float.
+     *
+     * @param fieldName name of the field
+     * @return the float value read
+     */
     readFloat(fieldName: string): number;
 
+    /**
+     * Reads a 16-bit signed integer.
+     *
+     * @param fieldName name of the field
+     * @return the short value read
+     */
     readShort(fieldName: string): number;
 
-    readPortable(fieldName: string): Portable;
+    /**
+     * Reads a Portable.
+     *
+     * @param fieldName name of the field
+     * @return the Portable value read
+     */
+    readPortable(fieldName: string): Portable | null;
 
-    readByteArray(fieldName: string): Buffer;
+    /**
+     * Reads a decimal.
+     *
+     * @param fieldName name of the field
+     * @return the BigDecimal value read
+     */
+    readDecimal(fieldName: string): BigDecimal | null;
 
-    readBooleanArray(fieldName: string): boolean[];
+    /**
+     * Reads a time.
+     *
+     * @param fieldName name of the field
+     * @return the LocalTime value read
+     */
+    readTime(fieldName: string): LocalTime | null;
 
-    readCharArray(fieldName: string): string[];
+    /**
+     * Reads a date.
+     *
+     * @param fieldName name of the field
+     * @return the LocalDate value read
+     */
+    readDate(fieldName: string): LocalDate | null;
 
-    readIntArray(fieldName: string): number[];
+    /**
+     * Reads a timestamp.
+     *
+     * @param fieldName name of the field
+     * @return the LocalDateTime value read
+     */
+    readTimestamp(fieldName: string): LocalDateTime | null;
 
-    readLongArray(fieldName: string): Long[];
+    /**
+     * Reads a timestampWithTimezone.
```

Comment:
```suggestion
     * Reads a timestamp with timezone.
```
to be consistent with the documentation of the write method